### PR TITLE
Optional parameter names.

### PR DESCRIPTION
**Pull request by LianaHus**
Now the names of input/return parameters of function do not have to be specified.

- added appropriate tests

### CODE REVIEW

File: libsolidity/AST.cpp [hunk 1]
```diff
@@ -58,10 +58,21 @@ void ContractDefinition::checkTypeRequirements()
 		BOOST_THROW_EXCEPTION(constructor->getReturnParameterList()->createTypeError(
 									"Non-empty \"returns\" directive for constructor."));
 
-	FunctionDefinition const* fallbackFunction = getFallbackFunction();
-	if (fallbackFunction && fallbackFunction->getScope() == this && !fallbackFunction->getParameters().empty())
-		BOOST_THROW_EXCEPTION(fallbackFunction->getParameterList().createTypeError(
-									"Fallback function cannot take parameters."));
+	FunctionDefinition const* fallbackFunction = nullptr;
+	for (ASTPointer<FunctionDefinition> const& function: getDefinedFunctions())
+	{
```
Comment (gavofyork) [inline libsolidity/AST.cpp]: remove unneeded `{`s.

**Comment by gavofyork** [comment]
looks good apart from minor style alteration.

File: libsolidity/Parser.cpp [hunk 2]
```diff
@@ -268,17 +268,28 @@ ASTPointer<VariableDeclaration> Parser::parseVariableDeclaration(VarDeclParserOp
...
 		isIndexed = true;
 		m_scanner->next();
 	}
-	Declaration::Visibility visibility(Declaration::Visibility::DEFAULT);
-	if (_options.isStateVariable && Token::isVisibilitySpecifier(token))
-		visibility = parseVisibilitySpecifier(token);
+	if (_options.allowEmptyName && m_scanner->getCurrentToken() != Token::IDENTIFIER)
+	{
+		identifier = make_shared<ASTString>("");
+		nodeFactory.setEndPositionFromNode(type);
```
Comment (chriseth) [inline libsolidity/Parser.cpp]: Unfortunately, this does not set the end position correctly... Yes, it is a really crappy mechanism.
I think we have to call it multiple times: Once `setEndPositionFromNode(type)` right after parsing the type and then `markEndPosition` at the beginning of the two if bodies (for visibility and indexed).

**Comment by gavofyork** [comment]
All parameters should be named when it's non-private; otherwise the JS side of the API will get confused.

**Comment by gavofyork** [comment]
also no compile on os x....

**Comment by chriseth** [comment]
@gavofyork Is that a temporary restriction, i.e. is it planned to add this feature to the JS side or is it impossible?
Also, is it ok to have anonymous return parameters or should parameters and return parameters be named?

**Comment by chriseth** [comment]
Seems to need a rebase again...

**Pull request merged**